### PR TITLE
fix: remediate erroneous white space in game and ach shortcode rendering

### PR DESCRIPTION
**Pull request by wescopeland**
Resolves https://github.com/RetroAchievements/RAWeb/issues/1509.

This is caused by the browser not trimming the extra whitespace when the shortcodes are rendered in a `<pre>` tag.

**Before**
(notice the space between the console name and game name on the first entry)
Screenshot 2023-05-01 at 11 11 01 PM

Screenshot 2023-05-02 at 1 54 50 PM

**After**
Screenshot 2023-05-01 at 11 10 24 PM

Screenshot 2023-05-02 at 1 54 41 PM

### CODE REVIEW

File: app_legacy/Helpers/render/game.php [hunk 1]
```diff
@@ -81,7 +81,7 @@ function renderGameTitle(?string $title = null, bool $tags = true): string
             . "<span class='tag-arrow'></span>"
             . "<span>$subset</span>"
             . "</span>";
-        $updateHtml($html, $matches[0], $tags ? " $span" : '');
+        $updateHtml($html, " $matches[0]", $tags ? " $span" : '');
```
Comment (Jamiras) [inline app_legacy/Helpers/render/game.php:84]: shouldn't the space be part of the match? what if the name doesn't have a preceding space?

(something maybe like this [untested])
`/\[Subset - (.+)\]/` => `/\s?\[Subset - (.+)\]/`

File: app_legacy/Helpers/render/achievement.php [hunk 2]
```diff
@@ -79,7 +79,11 @@ function renderAchievementTitle(?string $title, bool $tags = true): string
     }
     $title = str_replace('[m]', '', $title);
 
-    return "$title$missableTag";
+    // If we don't strip consecutive spaces, the
+    // browser doesn't collapse them in forum <pre> tags.
+    $title = preg_replace('/\s+/', ' ', $title);
```
Comment (Jamiras) [inline app_legacy/Helpers/render/achievement.php:84]: Does this actually do anything? Won't the whitespace just be a single preceding or trailing space (depending on where the [m] was)? If so, wouldn't `trim()` be easier?

Comment (wescopeland) [reply app_legacy/Helpers/render/achievement.php:84]: With the line removed:
Screenshot 2023-05-05 at 2 20 51 PM

With the line changed to `$title = trim($title);`:
Screenshot 2023-05-05 at 2 21 12 PM

With the line restored:
Screenshot 2023-05-05 at 2 21 23 PM

Comment (Jamiras) [reply app_legacy/Helpers/render/achievement.php:84]: Are the points already part of the title at this point? That's unfortunate.

Comment (wescopeland) [reply app_legacy/Helpers/render/achievement.php:84]: Based on my interpretation of what's happening in `achievementsAvatar()`, yes that appears to be the case.

**Review by Jamiras — APPROVED** [review]

**Pull request merged**